api/geth_backend.go

```diff
@@ -1289,32 +1289,34 @@ func (b *GethStatusBackend) ConnectionChange(typ string, expensive bool) {
 // AppStateChange handles app state changes (background/foreground).
 // state values: see https://facebook.github.io/react-native/docs/appstate.html
 func (b *GethStatusBackend) AppStateChange(state string) {
+	var messenger *protocol.Messenger
 	s, err := parseAppState(state)
 	if err != nil {
 		log.Error("AppStateChange failed, ignoring", "error", err)
 		return // and do nothing
 	}
 
-	b.log.Info("App State changed", "new-state", s)
 	b.appState = s
 
-	if b.statusNode != nil {
-		wakuext := b.statusNode.WakuExtService()
+	if b.statusNode.WakuV2ExtService() != nil {
+		messenger = b.statusNode.WakuV2ExtService().Messenger()
+	}
 
-		if wakuext != nil {
-			messenger := wakuext.Messenger()
+	if b.statusNode.WakuExtService() != nil {
+		messenger = b.statusNode.WakuExtService().Messenger()
+	}
 
-			if messenger != nil {
-				if s == appStateForeground {
-					messenger.ToForeground()
-				} else {
-					messenger.ToBackground()
-				}
+	if b.statusNode != nil {
```

Comment:
The nil checking needs to be before you use it,  you are checking it after you use it, you can use a guard pattern at the beginning:
Top of the function:
```

if b.statusNode == nil {
   return...
}
```
and remove the nil checking, similarly with messenger, just after you think you set it

```
if messenger == nil {
log.Warn("blah")
return ...
}
if s == appStateForeground  {
m.ToF.
} else {
m.ToBack
}
```

Reply:
Addressed here : https://github.com/status-im/status-go/pull/3189/commits/07205b79a2cb16c95e1742e230b928770103dca6